scripts/build.py

```diff
@@ -78,33 +88,77 @@ def process_third_party_libs():
         sys.exit(1)
 
 
+def copy_files_source_to_target(source, target):
+    """Copies all files in source directory to target."""
+    print 'Processing %s' % os.path.join(os.getcwd(), source)
+    print 'Copying into %s' % os.path.join(os.getcwd(), target)
+
+    ensure_directory_exists(target)
+    shutil.rmtree(target)
+    for root, dirs, files in os.walk(os.path.join(os.getcwd(), source)):
+        for directory in dirs:
+            print 'Processing %s' % os.path.join(root, directory)
+
+        for filename in files:
+            print 'Processing %s' % filename
+            source_path = os.path.join(root, filename)
+            if source_path.find(target) > 0:
+                continue
+            if source_path.find(source) == -1:
+                continue
+            target_path = source_path.replace(
+                source, target)
+            process_build(source_path, target_path)
+
+
 def _build_files():
     ensure_directory_exists(OUT_DIR)
     shutil.rmtree(OUT_DIR)
     process_third_party_libs()
 
-    for root in os.listdir(os.path.join(os.getcwd())):
-        if any([s in root for s in ['.git', 'third_party', 'extensions']]):
-            continue
-
-        print 'Processing %s' % os.path.join(os.getcwd(), root)
-        for root, dirs, files in os.walk(os.path.join(os.getcwd(), root)):
-            for directory in dirs:
-                print 'Processing %s' % os.path.join(root, directory)
-            for filename in files:
-                source_path = os.path.join(root, filename)
-                if source_path.find(OUT_DIR) > 0:
-                    continue
-                if source_path.find(HEAD_DIR) == -1:
-                    continue
-                target_path = source_path.replace(HEAD_DIR, OUT_DIR)
-                if filename.endswith('.html'):
-                    process_html(source_path, target_path)
-                if filename.endswith('.css'):
-                    process_css(source_path, target_path)
-                if filename.endswith('.js'):
-                    process_js(source_path, target_path)
+    for root, dirs, files in os.walk(os.path.join(os.getcwd(), HEAD_DIR)):
+        for directory in dirs:
+            print 'Processing %s' % os.path.join(root, directory)
+        for filename in files:
+            source_path = os.path.join(root, filename)
+            if source_path.find(OUT_DIR) > 0:
+                continue
+            if source_path.find(HEAD_DIR) == -1:
+                continue
+            target_path = source_path.replace(HEAD_DIR, OUT_DIR)
+            if filename.endswith('.html'):
+                process_html(source_path, target_path)
+            if filename.endswith('.css'):
+                process_css(source_path, target_path)
+            if filename.endswith('.js'):
+                process_js(source_path, target_path)
 
 
 if __name__ == '__main__':
+    # TODO(gvishal): get this cache slug somehow, unable to import the module.
```

Comment:
That's fine too, I think; alternatively, it should be simple enough to just "import yaml" and do "with open(file, 'r') as f:" and read the file's contents. You can define a new function in this module to do that. It doesn't necessarily have to be the same as the one in utils.py.